composeApp/src/commonMain/kotlin/presentation/screens/feed/FeedScreen.kt

```diff
@@ -0,0 +1,270 @@
+import androidx.compose.foundation.pager.HorizontalPager
+import androidx.compose.foundation.pager.rememberPagerState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.Icon
+import androidx.compose.material.ListItem
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Email
+import androidx.compose.material.icons.outlined.Favorite
+import androidx.compose.material.icons.outlined.MoreVert
+import androidx.compose.material.icons.outlined.Share
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.NavController
+import coil3.compose.AsyncImage
+import org.koin.compose.viewmodel.koinViewModel
+import presentation.NavigationRoutes.Detail
+import presentation.screens.feed.model.Post
+
+@Composable
+fun FeedScreen(
+    navController: NavController,
+    viewModel: FeedViewModel = koinViewModel()
+) {
+    val uiState by viewModel.uiState.collectAsState()
+
+    FeedScreenContent(
+        uiState,
+        openPost = { postId ->
+            navController.navigate("${Detail}/$postId")
+        },
+        onSharePost = { post ->
+            viewModel.onSharePost(post)
+        }
+    )
+}
+
+@Composable
+private fun FeedScreenContent(
+    uiState: FeedUiState,
+    openPost: (String) -> Unit = {},
+    onSharePost: (Post) -> Unit = {}
+) {
+    LazyColumn(
+        modifier = Modifier.fillMaxSize(),
+    ) {
+        items(uiState.posts) {
+            ItemPost(
+                post = it,
+                onClick = {
+                    openPost(it.id)
+                },
+                onShare = {
+                    onSharePost(it)
+                }
+            )
+        }
+    }
+}
+
+@Composable
+fun ItemPost(
+    post: Post,
+    onClick: () -> Unit = {},
+    onShare: () -> Unit = {}
+) {
+    PostHeader(post = post)
+    Spacer(modifier = Modifier.height(8.dp))
+    if (post.images.size > 1) {
+        PostMedia(images = post.images) {
+            onClick()
+        }
+    }
+    PostActions(
+        post = post,
+        onShare = onShare
+    )
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+private fun PostHeader(post: Post) {
```

Comment:
**#suggestion**

We can move this component to be a new file inside of a components package, just to keep the Screen file smaller.